oauth2_provider/views/mixins.py

```diff
@@ -155,9 +155,17 @@ def error_response(self, error, **kwargs):
         :param error: :attr:`OAuthToolkitError`
         """
         oauthlib_error = error.oauthlib_error
+
+        separator = '?'
+        try:
+            if '?' in oauthlib_error.redirect_uri:
+                separator = '&'
+        except:
+            pass
```

Comment:
sorry for being pedantic but I don't like `except: pass` very much ;)
What about something like:
```
redirect_uri = oauthlib_error.redirect_uri or ""
separator = '&' if '?' in redirect_uri else '?'
```
Also, I guess a test case checking the double ? problem (a test that fails without your patch) should be added